### ISSUE TITLE
haproxy as daemonset get oomkilled very quickly (and won't start)

### ISSUE THREAD

**Issue by Zempashi** [haproxytech/kubernetes-ingress#558]
Hello, 

On a staging environment with kube 1.27.4, installing haproxy kubernetes ingress controller with helm (actually via flux, so you have the version , the namespace, as extra info)
```
---                                        
apiVersion: helm.toolkit.fluxcd.io/v2beta1 
kind: HelmRelease
metadata: 
  name: ingress-haproxy
  namespace: ingress-system
spec:
  interval: 1h
  releaseName: ingress-haproxy
  chart:
    spec:
      sourceRef:
        kind: HelmRepository
        name: ingress-haproxy 
      version: 1.32.1
      chart: kubernetes-ingress
  values:
    controller:
      kind: DaemonSet
      nodeSelector:
        node-role.kubernetes.io/ingress: ""
      daemonset:
        useHostPort: true
      ingressClassResource:
        default: true
      publishService:
        enabled: false
      service:
        enabled: false
```

Then haproxy eat all memory available, then get oomkilled very quickly, get restarted, the oomkilled, etc...
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.                                                                                                                                                                                                                                                                                                                                                                                                                
[s6-init] ensuring user provided files have correct perms...exited 0.                                                                                                                                                                                                                                                                                                                                                                                                                        
[fix-attrs.d] applying ownership & permissions fixes...                                                                                                                                                                                                                                                                                                                                                                                                                                      
[fix-attrs.d] done.                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
[cont-init.d] executing container initialization scripts...                                                                                                                                                                                                                                                                                                                                                                                                                                  
[cont-init.d] 01-aux-cfg: executing...                                                                                                                                                                                                                                                                                                                                                                                                                                                       
[cont-init.d] 01-aux-cfg: exited 0.                                                                                                                                                                                                                                                                                                                                                                                                                                                          
[cont-init.d] done.                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
[services.d] starting services                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Memory limit for HAProxy: 0MiB                                                                                                                                                                                                                                                                                                                                                                                                                                                               
Memory limit for Ingress Controller: 0MiB                                                                                                                                                                                                                                                                                                                                                                                                                                                    
[services.d] done.                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
[NOTICE]   (212) : haproxy version is 2.6.9-3a3700a                                                                                                                                                                                                                                                                                                                                                                                                                                          
[WARNING]  (212) : config : missing timeouts for frontend 'https'.                                                                                                                                                                                                                                                                                                                                                                                                                           
   | While not properly invalid, you will certainly encounter various problems                                                                                                                                                                                                                                                                                                                                                                                                               
   | with such a configuration. To fix this, please ensure that all following                                                                                                                                                                                                                                                                                                                                                                                                                
   | timeouts are set to a non-zero value: 'client', 'connect', 'server'.                                                                                                                                                                                                                                                                                                                                                                                                                    
[WARNING]  (212) : config : missing timeouts for frontend 'http'.                                                                                                                                                                                                                                                                                                                                                                                                                            
   | While not properly invalid, you will certainly encounter various problems                                                                                                                                                                                                                                                                                                                                                                                                               
   | with such a configuration. To fix this, please ensure that all following                                                                                                                                                                                                                                                                                                                                                                                                                
   | timeouts are set to a non-zero value: 'client', 'connect', 'server'.                                                                                                                                                                                                                                                                                                                                                                                                                    
[WARNING]  (212) : config : missing timeouts for frontend 'healthz'.                                                                                                                                                                                                                                                                                                                                                                                                                         
   | While not properly invalid, you will certainly encounter various problems                                                                                                                                                                                                                                                                                                                                                                                                               
   | with such a configuration. To fix this, please ensure that all following                                                                                                                                                                                                                                                                                                                                                                                                                
   | timeouts are set to a non-zero value: 'client', 'connect', 'server'.                                                                                                                                                                                                                                                                                                                                                                                                                    
[WARNING]  (212) : config : missing timeouts for frontend 'stats'.                                                                                                                                                                                                                                                                                                                                                                                                                           
   | While not properly invalid, you will certainly encounter various problems                                                                                                                                                                                                                                                                                                                                                                                                               
   | with such a configuration. To fix this, please ensure that all following                                                                                                                                                                                                                                                                                                                                                                                                                
   | timeouts are set to a non-zero value: 'client', 'connect', 'server'.                                                                                                                                                                                                                                                                                                                                                                                                                    
[WARNING]  (212) : config : Removing incomplete section 'peers localinstance' (no peer named 'ingress-haproxy-kubernetes-ingress-mwwcj').                                                                                                                                                                                                                                                                                                                                                    
2023/07/24 09:03:22 maxprocs: Leaving GOMAXPROCS=1: CPU quota undefined                                                                                                                                                                                                                                                                                                                                                                                                                      
2023/07/24 09:03:22                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
 _   _    _    ____                                                                                                                                                                                                                                                                                                                                                                                                                                                                          
| | | |  / \  |  _ \ _ __ _____  ___   _                                                                                                                                                                                                                                                                                                                                                                                                                                                     
| |_| | / _ \ | |_) | '__/ _ \ \/ / | | |                                                                                                                                                                                                                                                                                                                                                                                                                                                    
|  _  |/ ___ \|  __/| | | (_) >  <| |_| |                                                                                                                                                                                                                                                                                                                                                                                                                                                    
|_| |_/_/   \_\_|   |_|  \___/_/\_\\__, |                                                                                                                                                                                                                                                                                                                                                                                                                                                    
 _  __     _                       |___/             ___ ____                                                                                                                                                                                                                                                                                                                                                                                                                                
| |/ /   _| |__   ___ _ __ _ __   ___| |_ ___  ___  |_ _/ ___|                                                                                                                                                                                                                                                                                                                                                                                                                               
| ' / | | | '_ \ / _ \ '__| '_ \ / _ \ __/ _ \/ __|  | | |                                                                                                                                                                                                                                                                                                                                                                                                                                   
| . \ |_| | |_) |  __/ |  | | | |  __/ ||  __/\__ \  | | |___                                                                                                                                                                                                                                                                                                                                                                                                                                
|_|\_\__,_|_.__/ \___|_|  |_| |_|\___|\__\___||___/ |___\____|                                                                                                                                                                                                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
2023/07/24 09:03:22 HAProxy Ingress Controller v1.9.5 24d2561.dirty                                                                                                                                                                                                                                                                                                                                                                                                                          
2023/07/24 09:03:22 Build from: https://github.com/haproxytech/kubernetes-ingress                                                                                                                                                                                                                                                                                                                                                                                                            
2023/07/24 09:03:22 Build date: 2023-03-10T13:12:58                                                                                                                                                                                                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
2023/07/24 09:03:22 ConfigMap: ingress-system/ingress-haproxy-kubernetes-ingress                                                                                                                                                                                                                                                                                                                                                                                                             
2023/07/24 09:03:22 Ingress class: haproxy                                                                                                                                                                                                                                                                                                                                                                                                                                                   
2023/07/24 09:03:22 Empty Ingress class: false                                                                                                                                                                                                                                                                                                                                                                                                                                               
2023/07/24 09:03:22 Publish service:                                                                                                                                                                                                                                                                                                                                                                                                                                                         
2023/07/24 09:03:22 Using local backend service on port: 6061                                                                                                                                                                                                                                                                                                                                                                                                                                
2023/07/24 09:03:22 Default ssl certificate: ingress-system/ingress-haproxy-kubernetes-ingress-default-cert                                                                                                                                                                                                                                                                                                                                                                                  
2023/07/24 09:03:22 Frontend HTTP listening on: 0.0.0.0:8080                                                                                                                                                                                                                                                                                                                                                                                                                                 
2023/07/24 09:03:22 Frontend HTTPS listening on: 0.0.0.0:8443                                                                                                                                                                                                                                                                                                                                                                                                                                
2023/07/24 09:03:22 Controller sync period: 5s                                                                                                                                                                                                                                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             
2023/07/24 09:03:22 Running on ingress-haproxy-kubernetes-ingress-mwwcj                                                                                                                                                                                                                                                                                                                                                                                                                      
2023/07/24 09:03:22 k8s/main.go:86 Running on Kubernetes version: v1.27.4 linux/amd64                                                                                                                                                                                                                                                                                                                                                                                                        
2023/07/24 09:03:22 INFO    k8s/main.go:155 Global CR defined in API core.haproxy.org/v1alpha2                                                                                                                                                                                                                                                                                                                                                                                               
2023/07/24 09:03:22 INFO    k8s/main.go:155 Defaults CR defined in API core.haproxy.org/v1alpha2                                                                                                                                                                                                                                                                                                                                                                                             
2023/07/24 09:03:22 INFO    k8s/main.go:155 Backend CR defined in API core.haproxy.org/v1alpha2                                                                                                                                                                                                                                                                                                                                                                                              
2023/07/24 09:03:22 INFO    builder.go:223 running default backend server on :6061                                                                                                                                                                                                                                                                                                                                                                                                           
2023/07/24 09:03:22 PANIC   builder.go:154 failed to initialize haproxy API client: dial unix /var/run/haproxy-runtime-api.sock: connect: no such file or directory                                                                                                                                                                                                                                                                                                                          
Error :  failed to initialize haproxy API client: dial unix /var/run/haproxy-runtime-api.sock: connect: no such file or directory
Memory limit for Ingress Controller: 0MiB                                                                                                                                                                                                     
2023/07/24 09:03:23 maxprocs: Leaving GOMAXPROCS=1: CPU quota undefined                                                                                                                                                                       
2023/07/24 09:03:23                                                                                                                                                                                                                           
 _   _    _    ____                                                                                                    
| | | |  / \  |  _ \ _ __ _____  ___   _                                                                                                                                                                                                      
| |_| | / _ \ | |_) | '__/ _ \ \/ / | | |                                                                                                                                                                                                     
|  _  |/ ___ \|  __/| | | (_) >  <| |_| |                      
|_| |_/_/   \_\_|   |_|  \___/_/\_\\__, |                      
 _  __     _                       |___/             ___ ____  
| |/ /   _| |__   ___ _ __ _ __   ___| |_ ___  ___  |_ _/ ___| 
| ' / | | | '_ \ / _ \ '__| '_ \ / _ \ __/ _ \/ __|  | | |      
| . \ |_| | |_) |  __/ |  | | | |  __/ ||  __/\__ \  | | |___ 
|_|\_\__,_|_.__/ \___|_|  |_| |_|\___|\__\___||___/ |___\____| 
                                                                                                                                                                                                                                              

2023/07/24 09:03:23 HAProxy Ingress Controller v1.9.5 24d2561.dirty
2023/07/24 09:03:23 Build from: https://github.com/haproxytech/kubernetes-ingress
2023/07/24 09:03:23 Build date: 2023-03-10T13:12:58                                                                    

2023/07/24 09:03:23 ConfigMap: ingress-system/ingress-haproxy-kubernetes-ingress
2023/07/24 09:03:23 Ingress class: haproxy 
[keep repeating same pattern]                                                                            
```

Tested with no ingress in the cluster, with a test ingress,

**Comment by dkorunic** [comment]
Ah, I have been able to reproduce this w/ kernel having keyword `max` in /sys/fs/cgroup/memory.max instead of an integer. I've sent an internal patch to remedy the handling of this situation accordingly and fallback to host calculation of MEMLIMIT.

**Comment by Zempashi** [comment]
For reference, the kernel is: `Linux node-1 6.4.4-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 19 Jul 2023 19:19:38 +0000 x86_64 GNU/Linux`

**Comment by dkorunic** [comment]
The fix for this issue is https://github.com/haproxytech/kubernetes-ingress/commit/f6685ce4719d9fbaffc61ec7ec37f2e267aab98a and it has been merged and released as [v1.10.6](https://github.com/haproxytech/kubernetes-ingress/releases/tag/v1.10.6).
